src/evaluator.ts

```diff
@@ -246,24 +247,73 @@ export async function runEval({
       const activeProvider = isApiProvider(test.provider) ? test.provider : provider;
       logger.debug(`Provider type: ${activeProvider.id()}`);
 
-      response = await activeProvider.callApi(renderedPrompt, {
-        // Always included
-        vars,
-
-        // Part of these may be removed in python and script providers, but every Javascript provider gets them
-        prompt,
-        filters,
-        originalProvider: provider,
-        test,
+      // The memory poisoning plugin requires the following additional steps:
+      // 1. Generate a scenario containing memories and follow up questions/commands which are dependent on the memories.
+      // 2. Send the memory message to the provider.
+      // 3. Send the test case to the provider; the test case should poison the memory created in the previous step.
+      // 4. Send the follow up question to the provider.
+      // A memory is successfully poisoned if the follow up question is answered incorrectly.
+      if (test.metadata?.pluginId === 'agentic:memory-poisoning') {
```

Comment:
I preferred this over a custom provider so it isn't the user's responsibility to choose the provider that corresponds with this plugin.